ci/docker_ci.sh

```diff
@@ -172,68 +136,221 @@ new_image_tag_name () {
     echo -n "${DOCKER_IMAGE_PREFIX}:${image_tag}"
 }
 
-# This prefix is altered for the private security images on setec builds.
-DOCKER_IMAGE_PREFIX="${DOCKER_IMAGE_PREFIX:-envoyproxy/envoy}"
-mkdir -p "${ENVOY_DOCKER_IMAGE_DIRECTORY}"
+build_platforms() {
+    local build_type=$1
 
-if is_windows; then
-  BUILD_TYPES=("-${WINDOWS_BUILD_TYPE}")
-  # BuildKit is not available for Windows images, use standard build command
-  BUILD_COMMAND=("build")
-else
-  # "-google-vrp" must come afer "" to ensure we rebuild the local base image dependency.
-  BUILD_TYPES=("" "-debug" "-contrib" "-contrib-debug" "-distroless" "-google-vrp" "-tools")
-
-  # Configure docker-buildx tools
-  BUILD_COMMAND=("buildx" "build")
-  if [[ -z "$DOCKER_CI_DRYRUN" ]]; then
-      config_env
-  fi
-fi
+    if is_windows; then
+        echo -n "windows/amd64"
+    elif [[ "${build_type}" == *-google-vrp ]]; then
+        echo -n "linux/amd64"
+    else
+        echo -n "linux/arm64,linux/amd64"
+    fi
+}
 
-# Test the docker build in all cases, but use a local tag that we will overwrite before push in the
-# cases where we do push.
-for BUILD_TYPE in "${BUILD_TYPES[@]}"; do
-  build_images "${BUILD_TYPE}" "$(image_tag_name "${BUILD_TYPE}")"
-
-  if ! is_windows; then
-      build_images "${BUILD_TYPE}" "$(new_image_tag_name "${BUILD_TYPE}")"
-  fi
-done
-
-# Only push images for main builds, and release branch builds
-if [[ "${AZP_BRANCH}" != "${MAIN_BRANCH}" ]] &&
-  ! [[ "${AZP_BRANCH}" =~ ${RELEASE_BRANCH_REGEX} ]]; then
-  echo 'Ignoring non-release branch for docker push.'
-  exit 0
-fi
+build_args() {
+    local build_type=$1 target
 
-if [[ -z "$DOCKER_CI_DRYRUN" ]]; then
-    docker login -u "$DOCKERHUB_USERNAME" -p "$DOCKERHUB_PASSWORD"
-fi
+    if is_windows; then
+        printf ' -f ci/Dockerfile-envoy-windows --build-arg BUILD_OS=%s --build-arg BUILD_TAG=%s' "${WINDOWS_IMAGE_BASE}" "${WINDOWS_IMAGE_TAG}"
+    else
+        target="${build_type/-debug/}"
+        target="${target/-contrib/}"
+        printf ' -f ci/Dockerfile-envoy --target %s' "envoy${target}"
+    fi
+
+    if [[ "${build_type}" == *-contrib* ]]; then
+        printf ' --build-arg ENVOY_BINARY=envoy-contrib'
+    fi
+
+    if [[ "${build_type}" == *-debug ]]; then
+        printf ' --build-arg ENVOY_BINARY_SUFFIX='
+    fi
+}
+
+use_builder() {
+    # BuildKit is not available for Windows images, skip this
+    if is_windows; then
+        return
+    fi
+    echo ">> BUILDX: use multi-builder"
+    echo "> docker buildx use multi-builder"
+
+    if [[ -n "$DOCKER_CI_DRYRUN" ]]; then
+        return
+    fi
+    docker buildx use multi-builder
+}
+
+build_and_maybe_push_image () {
+    # If the image is not required for local testing and this is main or a release this will push immediately
```

Comment:
... if it is required for testing on a main release branch it will push to a tar archive and then push to registry from there ...